project_close_forecast_date/models/project_project.py

```diff
@@ -0,0 +1,22 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, _
+from dateutil.relativedelta import relativedelta
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'last_close_date' in vals and vals['last_close_date']:
+            order_line_ids = self.env['purchase.order.line'].sudo().search([
+                ('account_analytic_id', 'in', self.analytic_account_id.ids),
+                ('order_id.invoice_status', '!=', 'invoiced')
```

Comment:
Well.. we might exclude cancelled lines IMO:
```suggestion
                ('order_id.invoice_status', '!=', 'invoiced'),
                ('state', '!=', 'cancel'),
```